airflow-core/src/airflow/serialization/serialized_objects.py

```diff
@@ -1135,7 +1135,8 @@ def detect_dag_dependencies(dag: DAG | None) -> Iterable[DagDependency]:
         """Detect dependencies set directly on the DAG object."""
         if not dag:
             return
-
+        if not dag.timetable:
+            return
```

Comment:
You shouldn't really end up with no timetable -- shouldn't you get a `NullTimetable` object instead?